rules/Php80/Rector/FunctionLike/UnionTypesRector.php

```diff
@@ -138,6 +162,56 @@ private function refactorParamTypes(FunctionLike $functionLike, PhpDocInfo $phpD
         }
     }
 
+    private function hasObjectWithoutClassType(UnionType $unionType): bool
+    {
+        $types = $unionType->getTypes();
+        foreach ($types as $type) {
+            if ($type instanceof ObjectWithoutClassType) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function hasObjectWithoutClassTypeWithOnlyFullyQualifiedObjectType(UnionType $unionType): bool
+    {
+        $types = $unionType->getTypes();
+        foreach ($types as $type) {
+            if ($type instanceof ObjectWithoutClassType) {
+                continue;
+            }
+
+            if (! $type instanceof FullyQualifiedObjectType) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private function cleanParamObjectType(
+        UnionType $unionType,
+        PhpDocInfo $phpDocInfo,
+        Param $param,
+        string $paramName
+    ): void
+    {
+        $types = $unionType->getTypes();
+        foreach ($types as $key => $type) {
+            if ($type instanceof ObjectWithoutClassType) {
+                unset($types[$key]);
+            }
+        }
+
+        $currentType = current($types);
+        $type = count($types) === 1
+            ? new ObjectType($currentType->getClassName())
```

Comment:
@TomasVotruba the type of `@param` with new ObjectType usage make : `DateTime` changed to `\DateTime`: 

with: 

```php
$this->phpDocTypeChanger->changeParamType($phpDocInfo, $type, $param, $paramName);
``` 

```diff
    /**
-     * @param DateTime $message
+     * @param \DateTime $message
     */
```

any idea to make it like "string" type with value "DateTime" ?

Reply:
@TomasVotruba implemented with `IdentifierTypeNode` 🎉